neat_ml/analysis/data_analysis.py

```diff
@@ -0,0 +1,747 @@
+    param: Optional[Union[int, float]] = None,
+) -> Dict[str, Any]:
+    """Builds a spatial graph from points and calculates network metrics.
+
+    Constructs a graph using Delaunay triangulation, radius search, or k-nearest
+    neighbors, then computes metrics like degree, clustering, and component sizes.
+
+    Parameters
+    ----------
+    points : Optional[np.ndarray]
+        An (N, 2) array of node coordinates.
+    areas : Optional[np.ndarray]
+        An (N,) array of node areas, used for Largest Connected
+        Component (LCC) area statistics.
+    method : str
+        The graph construction method: 'delaunay', 'radius', or 'knn'.
+    param : Optional[Union[int, float]]
+        The radius for 'radius' graphs or k for 'knn' graphs.
+
+    Returns
+    -------
+    Dict[str, Any]
+        A dictionary of graph metrics. Defaults to 0 for counts and NaN for
+        statistical measures if calculation is not possible.
+    """
+    baseline = {
+        "graph_num_nodes": 0,
+        "graph_num_edges": 0,
+        "graph_avg_degree": np.nan,
+        "graph_degree_std": np.nan,
+        "graph_num_components": np.nan,
+        "graph_lcc_node_fraction": np.nan,
+        "graph_avg_clustering": np.nan,
+        "graph_avg_neighbor_distance": np.nan,
+        "graph_avg_node_area_lcc": np.nan,
+    }
+
+    if (points is None or areas is None or points.shape[0] < 2 or
+            points.shape[0] != areas.shape[0]):
+        baseline["graph_num_nodes"] = 0 if points is None else int(points.shape[0])
+        return baseline
+
+    n_nodes = int(points.shape[0])
+    graph = nx.Graph()
+    graph.add_nodes_from((i, {"area": float(areas[i]), "pos": points[i]})
+                         for i in range(n_nodes))
+
+    try:
+        if method == "radius" and isinstance(param, (int, float)) and param > 0:
+            tree = KDTree(points)
+            for i, j in tree.query_pairs(r=float(param)):
+                dist = float(np.linalg.norm(points[i] - points[j]))
+                graph.add_edge(i, j, distance=dist)
+        elif method == "knn" and isinstance(param, int) and param > 0:
+            k = min(param, n_nodes - 1)
+            tree = KDTree(points)
+            dists, idxs = tree.query(points, k=k + 1)
+            for i in range(n_nodes):
+                for nn_idx in range(1, k + 1):
+                    j = int(idxs[i, nn_idx])
+                    if not graph.has_edge(i, j):
+                        dist = float(dists[i, nn_idx])
+                        graph.add_edge(i, j, distance=dist)
+        elif method == "delaunay" and n_nodes >= 3:
+            tri = Delaunay(points)
+            for simplex in tri.simplices:
+                for i, j in zip(simplex, np.roll(simplex, -1)):
+                    if not graph.has_edge(i, j):
+                        dist = float(np.linalg.norm(points[i] - points[j]))
+                        graph.add_edge(int(i), int(j), distance=dist)
+    except Exception as exc:
```

Comment:
Let's try to avoid capturing the generic `Exception` class for the usual reasons